### ISSUE TITLE
pernosco-submit asks for Mozilla ssh key passphrase multiple times

### ISSUE THREAD

**Issue by theres-waldo** [Pernosco/pernosco-submit#5]
```
$ ./pernosco-submit upload ~/.rr/latest-trace/
Running 'rr pack'...
rr: Packed trace directory `/home/botond/.rr/firefox-51'.
Copying /usr/lib/x86_64-linux-gnu/libthread_db.so into trace...
Obtaining source file list...
Enter passphrase for key '/home/botond/.ssh/id_mozilla':
Enter passphrase for key '/home/botond/.ssh/id_mozilla':
Enter passphrase for key '/home/botond/.ssh/id_mozilla':
```

Would be nicer if it only asked once.

**Comment by khuey** [comment]
I'm not sure why we're asking at all, the hg/git commands we run are pretty harmless. Do you have some strange configuration?

**Comment by theres-waldo** [comment]
I don't think so :)

What are the hg/git commands you run / is there a way for me see them?

**Comment by theres-waldo** [comment]
It looks like it's trying to compute the set of `outgoing()` revisions for each of my remotes.

One of my remotes is `ssh://hg.mozilla.org/integration/mozilla-inbound`, and performing that operation requires the ssh key.

Is that an unusual configuration my part?

**Comment by khuey** [comment]
If you pull the latest `pernosco-submit` `upload` now takes a `-x` that will echo commands.

Git at least supports separate push/fetch remotes, so you can use an authenticated remote for push but an unauthenticated one for fetch. I suspect Mercurial has something equivalent.

**Comment by theres-waldo** [comment]
It does, but Mercurial [deliberately](https://bz.mercurial-scm.org/show_bug.cgi?id=5365) uses the push URL for `outgoing()`.

**Comment by theres-waldo** [comment]
Also, just for completeness, the `-x` output is missing some quotes that would make the command runnable as writen :)

**Comment by theres-waldo** [comment]
From a user's point of view, my take on this is: why does pernosco care about **all** my remotes? I'd rather it let me specify a single remote (or a small subset) for the purpose of obtaining source files.

Note that performing operations on all remotes also adds latency to the `pernosco-submit` operation. For me, the "Obtaining source file list..." part of the operation takes 24 seconds.

**Comment by rocallahan** [comment]
> why does pernosco care about all my remotes? 

Because less manual configuration is better than more manual configuration, and there's no way to know which one to care about without manual configuration.

**Comment by rocallahan** [comment]
If the `outgoing(...)` revset lets us provide an explicit URL, we could probably use the 'pull' URL for each remote explicitly.

**Comment by rocallahan** [comment]
We could add some way to explicitly configure the remote to use, but it gets a bit complicated because you'd have to configure (remote, path) pairs since there can be multiple repositories involved.

**Comment by rocallahan** [comment]
> If the outgoing(...) revset lets us provide an explicit URL

It doesn't.

I suppose we could just ignore 'ssh:' URLs, and hope that all Mozilla users have at least one 'https:' URL configured.

**Comment by rocallahan** [comment]
Fixed: b36afb80055d57d5d9a7029e9dbf04f12a21d4a3
